src/LinuxTracing/PerfEventReaders.cpp

```diff
@@ -498,31 +431,26 @@ GenericTracepointPerfEvent ConsumeGenericTracepointPerfEvent(PerfEventRingBuffer
 
 SchedWakeupPerfEvent ConsumeSchedWakeupPerfEvent(PerfEventRingBuffer* ring_buffer,
                                                  const perf_event_header& header) {
-  ORBIT_CHECK(header.size >= sizeof(perf_event_raw_sample_fixed));
-  perf_event_raw_sample_fixed ring_buffer_record;
-  ring_buffer->ReadRawAtOffset(&ring_buffer_record, 0, sizeof(perf_event_raw_sample_fixed));
+  perf_event_attr flags{};
+
+  flags.sample_type = PERF_SAMPLE_RAW | SAMPLE_TYPE_TID_TIME_STREAMID_CPU;
+
+  PerfRecordSample res = ConsumeRecordSample(ring_buffer, header, flags);
 
-  // The last fields of the sched:sched_wakeup tracepoint aren't always the same, depending on the
-  // kernel version. Fortunately we only need the first fields, which are always the same, so only
-  // read those. See `sched_wakeup_tracepoint_fixed`.
-  ORBIT_CHECK(ring_buffer_record.size >= sizeof(sched_wakeup_tracepoint_fixed));
   sched_wakeup_tracepoint_fixed sched_wakeup;
-  ring_buffer->ReadRawAtOffset(
-      &sched_wakeup,
-      offsetof(perf_event_raw_sample_fixed, size) + sizeof(perf_event_raw_sample_fixed::size),
-      sizeof(sched_wakeup_tracepoint_fixed));
+  memcpy(&sched_wakeup, res.raw_data.get(), sizeof(sched_wakeup_tracepoint_fixed));
```

Comment:
Maybe a stupid question. But why you are only using memcpy here and not in the other methods?

Reply:
That data we get from `PERF_SAMPLE_RAW` is unstructured and is just an array of bytes.  `sched_wakeup_tracepoint_fixed` gives representation for that data and we copy it there to make it easier to work with. 
We need a `memcpy` everywhere we set `PERF_SAMPLE_RAW`.